src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitButton = cardModal.querySelector(".modal__submit-button");
+const cardModalCloseButton = cardModal.querySelector(".modal__close-button");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+
+// Avatar form elements
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form-avatar");
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarNameInput = avatarModal.querySelector("#profile-avatar-input");
+
+// delete modal elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector("#delete-form");
+const deleteModalCloseButton = deleteModal.querySelector(
+  ".modal__close-button"
+);
+const deleteModalCancelButton = deleteModal.querySelector(
+  ".modal__cancel-button"
+);
+
+// preview image popup elements
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = document.querySelector(".modal__image");
+const previewModalCaptionEl = document.querySelector(".modal__caption");
+const previewModalCloseButton = document.querySelector(
+  ".modal__close-button_type_preview"
+);
+
+// card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardList = document.querySelector(".cards__list");
+
+let selectedCard;
+let selectedCardId;
+
+/*
+function handleCardSubmit(evt) {
+  evt.preventDefault();
+  const values = { name: captionInput.value, link: cardLinkInput.value };
+  const cardEl = getCardElement(values);
+  cardList.prepend(cardEl);
+  closeModal(cardModal);
+  cardForm.reset();
+  disableButton(cardSubmitButton, validationConfig);
+}*/
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Delete", "Deleting...");
+    });
+}
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+function handleLike(evt, id) {
+  const button = evt.target;
+  const isLiked = button.classList.contains("card__like-button_liked");
+
+  api
+    .changeLikeStatus(id, isLiked)
+    .then((data) => {
+      button.classList.toggle("card__like-button_liked", data.isLiked);
+    })
+    .catch((err) => console.error(`Error updating like status: ${err}`));
+}
+
+function handleImageClick(data) {
+  openModal(previewModal);
+  const popupCaption = document.querySelector(".modal__caption");
```

Comment:
* There should be a `caption` under the image in the picture popup with the name of the image https://snipboard.io/y8W2A4.jpg